repositories/protectedRepository.js

```diff
@@ -0,0 +1,5 @@
+export class ProtectedRepository {
```

Comment:
Still don't quite see a reason for this repository

Reply:
how else should we differentiate calls that are protected with auth vs not protected?

Reply:
So the common thing is to use middleware that are tied to the routes. You write a middleware that checks auth headers, tokens etc and then you apply this middleware on the routes
Usage looks something like that - https://github.com/rainsxng/api_example/blob/master/src/routes/user.ts

Reply:
understood, thank you!
removed the ProtectedRepository, and adjusted the code according to the example!